### DISCUSSION TITLE
"Protocol not supported" error when trying to send emails

### DISCUSSION THREAD

**Discussion by danbee** [dani-garcia/vaultwarden#5159] [Q&A]
I'm trying to setup Vaultwarden to send emails and I'm running into some trouble. I have a valid SMTP server that I've tested independently so I know it works, but when I enter those details into the Vaultwarden config and get it to send an email I get the following errors in the log:

```
[2024-11-04 17:10:00.658][vaultwarden::mail][ERROR] SMTP error: Connection error: Protocol not supported (os error 43)
[2024-11-04 17:10:00.658][vaultwarden::api::core::accounts][ERROR] Error sending verify_email email: SMTP error: Connection error: Protocol not sup
ported (os error 43)
```

Vaultwarden is running in a jail on my TrueNAS (FreeBSD 13.1) server.

My SMTP server is Fastmail and I have tested it successfully so I know it works. I also tried SMTP2Go and got the same error.

Here are the relevant lines from my Vaultwarden config (with sensitive values scrubbed out):

```sh
SMTP_HOST=smtp.fastmail.com
export SMTP_HOST

SMTP_FROM=vaultwarden@mydomain.com
export SMTP_FROM

SMTP_PORT=465
export SMTP_PORT

SMTP_SECURITY=force_tls
export SMTP_SECURITY

SMTP_USERNAME={smtpusername}
export SMTP_USERNAME

SMTP_PASSWORD={smtppassword}
export SMTP_PASSWORD
```

**Comment by BlackDex** [comment]
Try the different options for this setting.

https://github.com/dani-garcia/vaultwarden/blob/2f20ad86f9283419845ff1b77c504c3dfd043578/.env.template#L528..L531

**Reply by BlackDex** [reply]
It might also be the port and security setting which is not supported or blocked in some way.

**Reply by danbee** [reply]
Yeah, the more I look into this the more I'm convinced my jail is broken in some way. Thanks for your response.
